tests/test_spark.py

```diff
@@ -430,3 +433,41 @@ def test_task_stdout():
         assert stdout_file["size"] > 0, "stdout file should have content"
     finally:
         sdk_install.uninstall(utils.SPARK_PACKAGE_NAME, service_name)
+
+
+@pytest.mark.sanity
+def test_handling_wrong_request_to_spark_dispatcher():
+    #submit a correct job
+    service_name = utils.FOLDERED_SPARK_SERVICE_NAME
+    utils.require_spark(service_name=service_name)
+    results  = sdk_cmd.run_raw_cli("dcos spark run --submit-args=\"--class org.apache.spark.examples.SparkPi https://downloads.mesosphere.com/spark/assets/spark-examples_2.11-2.0.1.jar 30\"")
+
+    #get it's submission id
+    submission_id = results.stdout.split(" ")[-1]
+
+    #submit a job through the REST interface with incorrect appArgs
+    sdk_cmd.run_raw_cli("node ssh --master-proxy --leader")
+    results = subprocess.run("hostname", shell=True, universal_newlines=True, check=True)
```

Comment:
not sure if this command will be actually executed on the `leader` host. In any case, we need to send REST request to Spark Dispatcher, not Mesos Master host.

Reply:
Modified approach.